extensions/amp-lightbox/0.1/amp-lightbox.js

```diff
@@ -181,10 +189,24 @@ class AmpLightbox extends AMP.BaseElement {
     this.action_ = Services.actionServiceForDoc(this.element);
     this.maybeSetTransparentBody_();
 
-    this.registerDefaultAction((i) => this.open_(i.trust), 'open');
-    this.registerAction('close', (i) => this.close(i.trust));
+    this.registerDefaultAction(i => this.open_(i.trust, i.caller), 'open');
+    this.registerAction('close', i => this.close(i.trust));
     /** If the element is in an email document, allow its `open` and `close` actions. */
     this.action_.addToWhitelist('AMP-LIGHTBOX', ['open', 'close'], ['email']);
+
+    this.maybeRenderCloseButtonHeader_();
+    this.closeButton_ = this.getExistingCloseButton_();
+    // If we do not have a close button provided by the page author, create one
+    // at the start of the lightbox for screen readers.
+    if (!this.closeButton_) {
+      this.closeButton_ = this.createScreenReaderCloseButton_();
```

Comment:
So the idea is to add this close button on top if there is not in the lightbox
And the line below is to create another one at the end so people using SR can close it when they are finished

Reply:
but maybe sometimes people don't want to have a close button 🤔 
I mean if the idea of the PR is improving accessibility maybe there is no need to add a close button for non accessibility reason and leave it as it is?

Reply:
So the idea is to send the focus in the lightbox because otherwise, a Screen reader (SR) user might never get to it. But you can't just focus on lightbox, it has to be something focussable. We also make it invisible so that's only for screen reader/focus
IF there is a focus in the lightbox (using autofocus or on tap focus): do nothing
ELSE IF there is a close button : focus on it
ELSE : create an invisible close button and focus on it

Reply:
so speaking about close buttons
We always want to have an invisible SR button in the end of the modal to make it possible for SR to focus it after reading the content. Right?
But why do we add a close button to the top if there is no close button?
Is it also needed for SR or it's a general lightbox problem?